caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -170,48 +174,51 @@ def __del__(self):
+        max_time: Optional[float] = None,
+        exponential_decay_length_penalty: Optional[Tuple[int, float]] = None,
+        stop_sequences: Optional[str] = None,
     ) -> GeneratedTextResult:
-        """Run the full text generation model.
-
-        Args:
-            text: str
-                Input string to be used to the generation model.
-            device: Optional[Union[str, int]]
-                Device on which we should run inference; by default, we use the detected device.
-            max_new_tokens: int
-                The maximum numbers of tokens to generate.
-                Default: 20
-            min_new_tokens: int
-                The minimum numbers of tokens to generate.
-                Default: 0 - means no minimum
+        self.run.__doc__ = """
+            Run the full text generation model.
+            Args:
+                {}
+            Returns:
+                GeneratedTextResult
+                    Generated text result produced by PEFT / Transformers.
+        """.format(
+            GENERATE_FUNCTION_ARGS
+        )
 
-        Returns:
-            GeneratedTextResult
-                Generated text result produced by PEFT / Transformers.
-        """
-        # Apply the verbalizer to our text string
         verbalized_text = render_verbalizer(self.verbalizer, {"input": text})
-        # Apply the tokenizer to the sample text & move to correct device
-        tok_tensors = self.tokenizer(verbalized_text, return_tensors="pt")
 
-        device = PeftPromptTuning._get_device(device)
-        inputs = {k: v.to(device) for k, v in tok_tensors.items()}
-        with torch.no_grad():
-            # Run tokenized tensors through the rest of the PEFT model
-            outputs = self.model.generate(
-                input_ids=inputs["input_ids"],
-                attention_mask=inputs["attention_mask"],
-                max_new_tokens=max_new_tokens,
-                min_new_tokens=min_new_tokens,
-                eos_token_id=self.eos_token_id,
-            )
-            gen_text = self.tokenizer.batch_decode(
-                outputs.detach().cpu().numpy(), skip_special_tokens=True
-            )
-        return GeneratedTextResult(generated_text=gen_text[0])
+        return generate_text_func(
+            self.model,
+            self.tokenizer,
+            self.PRODUCER_ID,
+            self.tokenizer.eos_token,
+            verbalized_text,
+            max_new_tokens=max_new_tokens,
+            min_new_tokens=min_new_tokens,
+            truncate_input_tokens=truncate_input_tokens,
+            decoding_method=decoding_method,
+            top_k=top_k,
+            top_p=top_p,
+            typical_p=typical_p,
+            temperature=temperature,
+            repetition_penalty=repetition_penalty,
+            max_time=max_time,
+            exponential_decay_length_penalty=exponential_decay_length_penalty,
+            stop_sequences=stop_sequences,
```

Comment:
What is the motivation for always passing the values here and setting our own defaults, rather than setting ours to None by default in `.run`, and only overriding the `not None` args (i.e., pulling the defaults from .generate instead)?

Reply:
In some places it was to align with TGIS (as per the proto def)

Reply:
Makes sense